src/event/input.ts

```diff
@@ -220,6 +222,35 @@ function calculateNewValue(
     }
   }
 
+  if (
+    isElementType(node, 'input', {type: 'number'} as const) &&
+    inputType === 'changeNumberInput' &&
+    !isDisabled(node) &&
+    !node.readOnly
+  ) {
+    const step = node.step ? Number(node.step) : 1
+
+    const reachedMax = value === node.max
+    if (inputData === 'ArrowUp' && !reachedMax) {
+      const exceedsMax = Number(value) + step > Number(node.max)
+      if (exceedsMax && !!node.max) {
+        newValue = node.max
+      } else {
+        newValue = (Number(value) + step).toString()
+      }
+    }
+
+    const reachedMin = value === node.min
+    if (inputData === 'ArrowDown' && !reachedMin) {
+      const exceedsMin = Number(value) - step < Number(node.min)
```

Comment:
same here... need to handle an empty value and bump to zero or the min value? or maybe max?